pkg/controller/podautoscaler/horizontal.go

```diff
@@ -367,7 +375,7 @@ func (a *HorizontalController) reconcileAutoscaler(hpav1Shared *autoscalingv1.Ho
 			desiredReplicas = calculateScaleUpLimit(currentReplicas)
 		}
 
-		rescale = shouldScale(hpa, currentReplicas, desiredReplicas, timestamp)
+		rescale = shouldScale(a, hpa, currentReplicas, desiredReplicas, timestamp)
```

Comment:
just make this a method of the HPA controller instead of passing the HPA as the first argument.